python/ray/data/dataset.py

```diff
@@ -318,88 +320,129 @@ def map_batches(
         fn_constructor_kwargs: Optional[Dict[str, Any]] = None,
         **ray_remote_args,
     ) -> "Dataset[Any]":
-        """Apply the given function to batches of records of this dataset.
+        """Apply the given function to batches of data.
 
-        The format of the data batch provided to ``fn`` can be controlled via the
-        ``batch_format`` argument, and the output of the UDF can be any batch type.
+        Batches are represented as dataframes, ndarrays, or lists. The default batch
+        type is determined by your dataset's schema. To determine the default batch
+        type, call :meth:`~Dataset.default_batch_format`. Alternatively, set the batch
+        type with ``batch_format``.
 
-        This is a blocking operation.
+        To learn more about writing functions for :meth:`~Dataset.map_batches`, read
+        :ref:`writing user-defined functions <transform_datasets_writing_udfs>`.
+
+        .. note::
+            This is a blocking operation. The time complexity is :math:`O(n/p)`, where
+            :math:`n` is the dataset size and :math:`p` is the parallelism.
+
+        .. tip::
+            If you're using :ref:`Ray AIR <air>` for training or batch inference,
+            consider using :class:`~ray.data.preprocessors.BatchMapper`. It's more
+            performant and easier to use.
 
         Examples:
+
+            >>> import pandas as pd
             >>> import ray
-            >>> # Transform python objects.
-            >>> ds = ray.data.range(1000)
-            >>> # Transform batches in parallel.
-            >>> ds.map_batches(lambda batch: [v * 2 for v in batch])
-            Dataset(num_blocks=..., num_rows=1000, schema=<class 'int'>)
-            >>> # Define a callable class that persists state across
-            >>> # function invocations for efficiency.
-            >>> init_model = ... # doctest: +SKIP
+            >>> df = pd.DataFrame({
+            ...     "name": ["Luna", "Rory", "Scout"],
+            ...     "age": [4, 14, 9]
+            ... })
+            >>> ds = ray.data.from_pandas(df)
+            >>> ds
+            Dataset(num_blocks=1, num_rows=3, schema={name: object, age: int64})
+
+            Call :meth:`~Dataset.default_batch_format` to determine the default batch
+            type.
```

Comment:
Nit: I think that a simple `.default_batch_format` reference should work, and is a bit more concise/readable in the non-rendered docs. `.default_batch_format` would include the `.` in the displayed text, while `~.default_batch_format` would remove it; I think that the former is better since it makes it more obvious that it's a method on the current class. Lmk what you think.
```suggestion
            Call :meth:`.default_batch_format` to determine the default batch type.
```

Reply:
When I rendered the documentation, the `.` wasn't included.

<img width="750" alt="image" src="https://user-images.githubusercontent.com/26107013/190521548-4e0675c6-19d2-4179-b1e7-3728b6316fb2.png">

In any case, `.default_batch_format` is definitely more concise, so I've applied the suggestion. Didn't know you could do that

Reply:
Hmm interesting, the `.` must be implicitly sliced off. Good to know!